pya/aserver.py

```diff
@@ -287,5 +292,18 @@ def _play_callback(self, in_data, frame_count, time_info, flag):
     def stop(self):
         self._stop = True
 
-    def __del__(self):
+    def _terminate_backend(self):
         self.backend.terminate()
+
+    def __enter__(self):
+        return self.boot()
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        if self._is_active:
+            self.quit()
```

Comment:
i'd rather put the 'active' check in `self.quit`. this way calling the function will handle side effects instead of making the caller responsible for it.